### PR TITLE
fix: apple provider

### LINKED ISSUE #490: [Social] Issue signin with apple
When I try to sign in with apple, I keep getting the following error: 
```invalid_request
response_mode must be form_post when name or email scope is requested.
```
image
this the authorization url:

```
https://appleid.apple.com/auth/authorize?client_id=XXX&response_type=code&redirect_uri=https://local.api.ambrosia.events/api/auth/callback/apple&scope=email%20name%20openid&state=XXX```

When I append `&response_mode=form_post` to the url manually, the error is no more. But then it fails on the callback url.

<img width="1343" alt="image" src="https://github.com/user-attachments/assets/654811e1-f9b0-44e8-afcb-ea7a40195092">

### PR DESCRIPTION
**Pull request by Bekacru**
closes #490

### CODE REVIEW

File: packages/better-auth/src/social-providers/apple.ts [hunk 1]
```diff
@@ -60,14 +73,14 @@ export const apple = (options: AppleOptions) => {
...
 		name: "Apple",
 		createAuthorizationURL({ state, scopes, redirectURI }) {
-			const _scope = scopes || ["email", "name", "openid"];
+			const _scope = scopes || ["email", "name"];
 			options.scope && _scope.push(...options.scope);
 			return new URL(
 				`https://appleid.apple.com/auth/authorize?client_id=${
 					options.clientId
-				}&response_type=code&redirect_uri=${
+				}&response_type=form_post&redirect_uri=${
```
Comment (emroot) [inline packages/better-auth/src/social-providers/apple.ts]: @Bekacru we wanna keep this as code. Form_post is for response_mode

**Comment by Bekacru** [comment]
@emroot could yout ry `0.8.4-beta.2` to see if it's working?

**Comment by emroot** [comment]
Yeah I could try and yes it in about an hour or so.

**Comment by Bekacru** [comment]
> Yeah I could try and yes it in about an hour or so.

sure

File: packages/better-auth/src/social-providers/apple.ts [hunk 2]
```diff
@@ -77,6 +90,7 @@ export const apple = (options: AppleOptions) => {
 				redirectURI: options.redirectURI || redirectURI,
 				options,
 				tokenEndpoint,
+				authentication: "basic",
```
Comment (emroot) [inline packages/better-auth/src/social-providers/apple.ts]: this line breaks the apple auth. When present I keep getting an error: "invalid_client".
Works when no present.

**Review by emroot — CHANGES_REQUESTED** [review]
Added a comment. LGTM when that is fixed.
Unrelated to this code, but I keep getting an error `unable_to_create_user`. Looks like it might be related to this https://github.com/better-auth/better-auth/pull/513. 
generateId is not respected anymore. I'm relaying on prisma to generate the ID, and setting `generateId: false` now doesn't work with prisma adapter.

**Comment by Bekacru** [comment]
> generateId is not respected anymore.

should be fixed on latest :)

**Pull request merged**